src/MLJInterface.jl

```diff
@@ -103,9 +106,57 @@ function get_options(::AbstractSRRegressor) end
 eval(modelexpr(:SRRegressor))
 eval(modelexpr(:MultitargetSRRegressor))
 
+"""
+    SRFitResultTypes
+
+A struct referencing types in the `SRFitResult` struct,
+to be used in type inference during MLJ.update to speed up iterative fits.
+"""
+Base.@kwdef struct SRFitResultTypes{
+    _T,_X_t,_y_t,_w_t,_state,_X_units,_y_units,_X_units_clean,_y_units_clean
+}
+    T::Type{_T} = Any
+    X_t::Type{_X_t} = Any
+    y_t::Type{_y_t} = Any
+    w_t::Type{_w_t} = Any
+    state::Type{_state} = Any
+    X_units::Type{_X_units} = Any
+    y_units::Type{_y_units} = Any
+    X_units_clean::Type{_X_units_clean} = Any
+    y_units_clean::Type{_y_units_clean} = Any
+end
+
+"""
+    SRFitResult
+
+A struct containing the result of a fit of an `SRRegressor` or `MultitargetSRRegressor`.
+"""
+Base.@kwdef struct SRFitResult{
+    M<:AbstractSRRegressor,
+    S,
+    O<:Options,
+    XD<:Union{Vector{<:AbstractDimensions},Nothing},
+    YD<:Union{AbstractDimensions,Vector{<:AbstractDimensions},Nothing},
+    TYPES<:SRFitResultTypes,
+}
+    model::M
+    state::S
+    num_targets::Int
+    options::O
+    variable_names::Vector{String}
+    y_variable_names::Union{Vector{String},Nothing}
+    y_is_table::Bool
+    has_class::Bool
+    X_units::XD
+    y_units::YD
+    types::TYPES
+end
```

Comment:
For some reason this is breaking `fit!(mach)` from being run twice... Need to investigate